packages/react-components/react-avatar/src/components/AvatarGroup/useAvatarGroupStyles.ts

```diff
@@ -18,34 +22,179 @@ const useStyles = makeStyles({
     display: 'inline-flex',
     position: 'relative',
   },
+  pie: {
+    clipPath: 'circle(50%)',
+  },
+});
+
+/**
+ * Styles for overflow button slot.
+ */
+const useOverflowButtonStyles = makeStyles({
+  base: {
+    display: 'inline-flex',
+    position: 'relative',
+    flexShrink: 0,
+    justifyContent: 'center',
+    alignItems: 'center',
+    color: tokens.colorNeutralForeground1,
+    backgroundColor: tokens.colorNeutralBackground1,
+    ...shorthands.borderWidth(`var(${avatarGroupOverflowButtonBorderVar})`),
+    ...shorthands.borderColor(tokens.colorNeutralStroke1),
+    ...shorthands.borderRadius(tokens.borderRadiusCircular),
+    ...shorthands.borderStyle('solid'),
+    ...shorthands.padding(tokens.spacingHorizontalNone),
+  },
+
+  // These styles match the default button styles
+  focusIndicator: createCustomFocusIndicatorStyle({
+    ...shorthands.borderColor('transparent'),
+    outlineColor: 'transparent',
+    outlineWidth: tokens.strokeWidthThick,
+    outlineStyle: 'solid',
+    boxShadow: `${tokens.shadow4} 0 0 0 2px ${tokens.colorStrokeFocus2}`,
+    zIndex: 1,
+  }),
+
+  states: {
+    ':hover': {
+      color: tokens.colorNeutralForeground1Hover,
+      backgroundColor: tokens.colorNeutralBackground1Hover,
+      ...shorthands.borderColor(tokens.colorNeutralStroke1Hover),
+    },
+    ':active': {
+      color: tokens.colorNeutralForeground1Pressed,
+      backgroundColor: tokens.colorNeutralBackground1Pressed,
+      ...shorthands.borderColor(tokens.colorNeutralStroke1Pressed),
+    },
+    ':checked': {
+      color: tokens.colorNeutralForeground1Selected,
+      backgroundColor: tokens.colorNeutralBackground1Selected,
+      ...shorthands.borderColor(tokens.colorNeutralStroke1Selected),
+    },
+  },
+
+  pie: {
+    backgroundColor: tokens.colorTransparentBackground,
+    ...shorthands.borderColor(tokens.colorTransparentStroke),
+    color: 'transparent',
+  },
+
+  stack: {
+    marginLeft: `var(${avatarGroupItemMarginVar})`,
+    outlineColor: tokens.colorNeutralBackground2,
+    outlineStyle: 'solid',
+    outlineWidth: `var(${avatarGroupItemOutlineVar})`,
+  },
```

Comment:
The reason for these variables is because I will use them in the `AvatarGroupItem` as well. Would that be an okay case for the variables?

Reply:
I think AvatarGroup should handle the layout of the items inside, rather than using a CSS variable and relying on AvatarGroupItem to set a margin.

Ideally this would be done with the columnGap property. Unfortunately we can't use that because older browsers don't support it. Instead, you could set a left margin on all children:
```
  stackXXS: {
    '& > *': { marginLeft: `calc(-1 * ${tokens.spacingHorizontalXXS})` },
  },
```

Or another appropriate selector (like using `& > *:not(:first-child)` or `& > * + *`), depending on what you need.